h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -658,10 +655,12 @@ private int readAnInt() throws IOException {
   /** Put as needed to keep from overflowing the ByteBuffer. */
   private ByteBuffer putSp( int sz ) {
     assert !_read;
-    while (sz > _bb.remaining()) {
```

Comment:
We don't actually need to loop here. expandByteBuffer() either can only modify the buffer in the first call, second and other calls are effectively NOOP. sendPartial() already has an inner loop that depletes the buffer and ends up with a clear buffer.